staging/src/k8s.io/kube-aggregator/pkg/cmd/server/start.go

```diff
@@ -55,21 +55,23 @@ func NewCommandStartAggregator(out, err io.Writer, stopCh <-chan struct{}) *cobr
 	cmd := &cobra.Command{
 		Short: "Launch a API aggregator and proxy server",
 		Long:  "Launch a API aggregator and proxy server",
-		RunE: func(c *cobra.Command, args []string) error {
-			if err := o.Complete(); err != nil {
-				return err
-			}
-			if err := o.Validate(args); err != nil {
-				return err
-			}
-			if err := o.RunAggregator(stopCh); err != nil {
-				return err
-			}
-			return nil
-		},
+	}
```

Comment:
where is the difference to the old setting RunE in the struct construction above?

Reply:
And why the comment?

Reply:
I move flag into RunE, otherwise if we parse flags in test, we will encounter flag redefined error.

Reply:
> And why the comment?

This is only for mock RunE, so we can pass listener into options

Reply:
Should we impl a fake one directly copy this code? I think this way is easy, and not change the original code.